src/Tasks/SystemState.cs

```diff
@@ -596,6 +606,74 @@ out fileState.frameworkName
             frameworkName = fileState.frameworkName;
         }
 
+        /// <summary>
+        /// Reads in cached data from stateFiles to build an initial cache. Avoids logging warnings or errors.
+        /// </summary>
+        /// <param name="stateFiles">List of locations of caches on disk.</param>
+        /// <param name="log">How to log</param>
+        /// <param name="fileExists">Whether a file exists</param>
+        /// <returns></returns>
+        internal static SystemState DeserializePrecomputedCachesByTranslator(ITaskItem[] stateFiles, TaskLoggingHelper log, FileExists fileExists)
+        {
+            SystemState retVal = new SystemState();
+            retVal.isDirty = stateFiles.Length > 0;
```

Comment:
isDirty shouldn't matter at this point in almost any case, since we'd presumably modify the cache before considering serializing it anyway unless there really is nothing of interest. I actually slightly prefer this version, since if there are invalid caches, we won't try repeatedly to read something invalid, instead reading something empty, which is less computationally expensive.